src/quantum/azext_quantum/operations/job.py

```diff
@@ -174,8 +194,201 @@ def _has_completed(job):
+            content_encoding = job_params["content_encoding"]
+            del job_params["content_encoding"]
+        if "return_sas_token" in job_params.keys():
+            return_sas_token_str = job_params["return_sas_token"]
+            return_sas_token = return_sas_token_str.lower() == "true"
+            del job_params["return_sas_token"]
+
+    # Prepare for input file upload according to job type
+    if job_type == QIO_JOB:
+        container_name_prefix = "cli-qio-job-"
+        if content_type is None:
+            content_type = "application/json"
+        if content_encoding is None:
+            content_encoding = "gzip"
+        if return_sas_token is None:
+            return_sas_token = True
+        with open(job_input_file, encoding="utf-8") as qio_file:
+            uncompressed_blob_data = qio_file.read()
+
+        if ("content_type" in uncompressed_blob_data and "application/x-protobuf" in uncompressed_blob_data) or (content_type.lower() == "application/x-protobuf"):
+            raise InvalidArgumentValueError('Content type "application/x-protobuf" is not supported.')
+
+        # Compress the input data (This code is based on to_blob in qdk-python\azure-quantum\azure\quantum\optimization\problem.py)
+        data = io.BytesIO()
+        with gzip.GzipFile(fileobj=data, mode="w") as fo:
+            fo.write(uncompressed_blob_data.encode())
+        blob_data = data.getvalue()
+
+    else:
+        if job_type == QIR_JOB:
+            container_name_prefix = "cli-qir-job-"
+            if content_type is None:
+                content_type = "application/x-qir.v1"       # <<<<< Is this a valid default for all QIR jobs?
+            content_encoding = None
+        else:
+            container_name_prefix = "cli-pass-through-job-"
+
+        if return_sas_token is None:
+            return_sas_token = False
+
+        with open(job_input_file, "rb") as input_file:
+            blob_data = input_file.read()
+
+    # Upload the input file to the workspace's storage account
+    if storage is None:
+        from .workspace import get as ws_get
+        ws = ws_get(cmd)
+        storage = ws.storage_account.split('/')[-1]
+    job_id = str(uuid.uuid4())
+    container_name = container_name_prefix + job_id
+    connection_string_dict = show_storage_account_connection_string(cmd, resource_group_name, storage)
+    connection_string = connection_string_dict["connectionString"]
+    container_client = create_container(connection_string, container_name)
+    blob_name = "inputData"
+    knack_logger.warning('Uploading input data...')
+    blob_uri = upload_blob(container_client, blob_name, content_type, content_encoding, blob_data, return_sas_token)
+
+    # Remove blob name from container URI
+    start_of_blob_name = blob_uri.find(blob_name)
+    if return_sas_token:
+        end_of_blob_name = blob_uri.find("?")
+        container_uri = blob_uri[0:start_of_blob_name - 1] + blob_uri[end_of_blob_name:]
```

Comment:
wonder if you can share a sample container uri?

Reply:
Removed the return_sas_token flag.  Container URIs will never contain SAS tokens.